bundles/org.openhab.binding.intesis/src/main/java/org/openhab/binding/intesis/internal/handler/IntesisHomeHandler.java

```diff
@@ -0,0 +1,537 @@
+
+            });
+        }
+    }
+
+    public @Nullable String login() {
+        String contentString = "{\"command\":\"login\",\"data\":{\"username\":\"Admin\",\"password\":\"" + password
+                + "\"}}";
+        String response = api.postRequest(ipAddress, contentString);
+        String sessionId = null;
+        if (response != null) {
+            try {
+                Response resp = gson.fromJson(response, Response.class);
+                boolean success = resp.success;
+                if (success) {
+                    Data data = gson.fromJson(resp.data, Data.class);
+                    Id id = gson.fromJson(data.id, Id.class);
+                    sessionId = id.sessionID.toString();
+                    if (sessionId != null && !sessionId.isEmpty()) {
+                        updateStatus(ThingStatus.ONLINE);
+                        return sessionId;
+                    } else {
+                        updateStatus(ThingStatus.OFFLINE);
+                    }
+                } else {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+                    return null;
+                }
+            } catch (JsonSyntaxException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            }
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+            return null;
+        }
+        return sessionId;
+    }
+
+    public @Nullable String logout(String sessionId) {
+        String contentString = "{\"command\":\"logout\",\"data\":{\"sessionID\":\"" + sessionId + "\"}}";
+        String response = api.postRequest(ipAddress, contentString);
+        return response;
+    }
+
+    public void populateProperties() {
+        String contentString = "{\"command\":\"getinfo\",\"data\":\"\"}";
+        String response = api.postRequest(ipAddress, contentString);
+        logger.trace("getInfo response : {}", response);
+        if (response != null) {
+            try {
+                Response resp = gson.fromJson(response, Response.class);
+                boolean success = resp.success;
+                if (success) {
+                    Data data = gson.fromJson(resp.data, Data.class);
+                    Info info = gson.fromJson(data.info, Info.class);
+                    properties.put(PROPERTY_VENDOR, "Intesis");
+                    properties.put(PROPERTY_MODEL_ID, info.deviceModel);
+                    properties.put(PROPERTY_SERIAL_NUMBER, info.sn);
+                    properties.put(PROPERTY_FIRMWARE_VERSION, info.fwVersion);
+                    properties.put(PROPERTY_MAC_ADDRESS, info.wlanSTAMAC);
+                    updateStatus(ThingStatus.ONLINE);
+                }
+            } catch (JsonSyntaxException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+            }
+        }
```

Comment:
This code can now also be refactored to use our new `postRequest` method.
```suggestion
        postRequest("{\"command\":\"getinfo\",\"data\":\"\"}", resp -> {
                    Data data = gson.fromJson(resp.data, Data.class);
                    Info info = gson.fromJson(data.info, Info.class);
                    properties.put(PROPERTY_VENDOR, "Intesis");
                    properties.put(PROPERTY_MODEL_ID, info.deviceModel);
                    properties.put(PROPERTY_SERIAL_NUMBER, info.sn);
                    properties.put(PROPERTY_FIRMWARE_VERSION, info.fwVersion);
                    properties.put(PROPERTY_MAC_ADDRESS, info.wlanSTAMAC);
                    updateStatus(ThingStatus.ONLINE);
        });
```

Reply:
Done